src/StackExchange.Redis/RedisTransaction.cs

```diff
@@ -117,20 +117,23 @@ private void QueueMessage(Message message)
             {
                 (_pending ??= new List<QueuedMessage>()).Add(queued);
 
-                switch (message.Command)
```

Comment:
can we move the test *inside* the `switch`? I'd rather not execute the `IsAvailable` check etc unless we think there's a good need; other than that: makes sense

Reply:
not related to this change, but: unless I'm misreading the code, it *looks* like we'd issue this even on "cluster"; open question: should we also add a single-DB check? we usually know the DB count, either from server metadata (when available) or from the server type

Reply:
sounds good wrt to rearranging things.

WRT preventing an issue with cluster - what do you think of this new approach?